pkg/cmd/workflow/run/run.go

```diff
@@ -0,0 +1,408 @@
+
+func (ia *InputAnswer) WriteAnswer(name string, value interface{}) error {
+	if s, ok := value.(string); ok {
+		ia.providedInputs[name] = s
+		return nil
+	}
+
+	// TODO i hate this; this is to make tests work:
+	if rv, ok := value.(reflect.Value); ok {
+		ia.providedInputs[name] = rv.String()
+		return nil
+	}
+
+	return fmt.Errorf("unexpected value type: %v", value)
+}
+
+func collectInputs(yamlContent []byte) (map[string]string, error) {
+	inputs, err := findInputs(yamlContent)
+	if err != nil {
+		return nil, err
+	}
+
+	providedInputs := map[string]string{}
+
+	if len(inputs) == 0 {
+		return providedInputs, nil
+	}
+
+	qs := []*survey.Question{}
+	for inputName, input := range inputs {
+		q := &survey.Question{
+			Name: inputName,
+			Prompt: &survey.Input{
+				Message: inputName,
+				Default: input.Default,
+			},
+		}
+		if input.Required {
+			q.Validate = survey.Required
+		}
+		qs = append(qs, q)
+	}
+
+	sort.Slice(qs, func(i, j int) bool {
+		return qs[i].Name < qs[j].Name
+	})
+
+	inputAnswer := InputAnswer{
+		providedInputs: providedInputs,
+	}
+	err = prompt.SurveyAsk(qs, &inputAnswer)
+	if err != nil {
+		return nil, err
+	}
+
+	return providedInputs, nil
+}
+
+func runRun(opts *RunOptions) error {
+	c, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("could not build http client: %w", err)
+	}
+	client := api.NewClientFromHTTP(c)
+
+	repo, err := opts.BaseRepo()
+	if err != nil {
+		return fmt.Errorf("could not determine base repo: %w", err)
+	}
+
+	ref := opts.Ref
+
+	if ref == "" {
+		ref, err = api.RepoDefaultBranch(client, repo)
+		if err != nil {
+			return fmt.Errorf("unable to determine default branch for %s: %w", ghrepo.FullName(repo), err)
+		}
+	}
+
+	states := []shared.WorkflowState{shared.Active}
+	workflow, err := shared.ResolveWorkflow(
+		opts.IO, client, repo, opts.Prompt, opts.Selector, states)
+	if err != nil {
+		var fae shared.FilteredAllError
+		if errors.As(err, &fae) {
+			return errors.New("no workflows are enabled on this repository")
+		}
+		return err
+	}
+
+	providedInputs := map[string]string{}
+
+	if len(opts.MagicFields)+len(opts.RawFields) > 0 {
+		providedInputs, err = parseFields(*opts)
+		if err != nil {
+			return err
+		}
+	}
+
+	if opts.JSON != "" {
+		err := json.Unmarshal([]byte(opts.JSON), &providedInputs)
+		if err != nil {
+			return fmt.Errorf("could not parse provided JSON: %w", err)
+		}
+	}
+
+	if opts.Prompt {
+		yamlContent, err := shared.GetWorkflowContent(client, repo, *workflow, ref)
+		if err != nil {
+			return fmt.Errorf("unable to fetch workflow file content: %w", err)
+		}
+		providedInputs, err = collectInputs(yamlContent)
+		if err != nil {
+			return err
+		}
+	}
+
+	path := fmt.Sprintf("repos/%s/actions/workflows/%d/dispatches",
+		ghrepo.FullName(repo), workflow.ID)
+
+	requestByte, err := json.Marshal(struct {
+		Ref    string            `json:"ref"`
+		Inputs map[string]string `json:"inputs"`
+	}{
+		Ref:    ref,
+		Inputs: providedInputs,
+	})
```

Comment:
Instead of defining a new struct type inline, I think that passing a simple map would be enough!
```suggestion
	requestByte, err := json.Marshal(map[string]interface{}{
		"ref":    ref,
		"inputs": providedInputs,
	})
```